tests/test_core.py

```diff
@@ -7383,15 +7391,18 @@ def make_run(fullname, name=-1, compiler=-1, embetter=0, quantum_size=0,
   TT = type(fullname, (T,), dict(run_name = fullname, env = env))
 
   def tearDown(self):
-    super(TT, self).tearDown()
-
-    for k, v in self.env.iteritems():
-      del os.environ[k]
+    try:
+      super(TT, self).tearDown()
+    except Exception, e:
```

Comment:
LGTM except for something here. It looks like you're catching an exception if there is one, and continuing normally? Is that intentional? I would have thought to note the exception, do the cleanup in the finally, then re-throw the exception if there was one?